src/socket/address.cr

```diff
@@ -179,20 +179,46 @@ class Socket
   # Holds the local path of an UNIX address, usually coming from an opened
   # connection (e.g. `Socket#local_address`, `Socket#receive`).
   #
+  # You may also declare an abstract UNIX address, that is a virtual file
+  # that will never be created on the filesystem. An abstract UNIX address
+  # path is prefixed by a `@` character.
+  #
+  # NOTE:
+  # - Abstract UNIX addresses are supported only on some Linux systems.
+  # - If you want to use a non-abstract UNIX address starting with a `@`,
+  # prefix it with `./`, like `./@foo`.
+  #
   # Example:
   # ```
   # Socket::UNIXAddress.new("/tmp/my.sock")
+  #
+  # # Abstract UNIX socket on Linux only
+  # Socket::UNIXAddress.new("@/my.sock")
   # ```
   struct UNIXAddress < Address
     getter path : String
+    getter? abstract : Bool
 
     # :nodoc:
     MAX_PATH_SIZE = LibC::SockaddrUn.new.sun_path.size - 1
 
-    def initialize(@path : String)
-      if @path.bytesize + 1 > MAX_PATH_SIZE
+    def initialize(path : String)
+      if path.bytesize + 1 > MAX_PATH_SIZE
         raise ArgumentError.new("Path size exceeds the maximum size of #{MAX_PATH_SIZE} bytes")
       end
+
+      if path[0]? == '@'
```

Comment:
`path.starts_with?('@')`

Reply:
yeah I didn't want to do that because unconsciously for me `starts_with?` is to check that it starts with a `String`, not a `Char`....

Nevermind I'll change it.